### DISCUSSION TITLE
How to merge multiple MAF files into one MAF file ?

### DISCUSSION THREAD

**Discussion by GAUSS66-well** [PoisonAlien/maftools#822] [Q&A] [answered]
Hello, I'm a rookie of R language. I want to merge multiple MAF files into one MAF file and export table or txt file  to my computer. All MAFS are in one folder. What should I do? thanks a lot.

![image](https://user-images.githubusercontent.com/77564128/163246761-1cbffebd-870f-47b0-bb7f-32a4e2fd43e9.png)

**Comment by PoisonAlien** [comment, marked as answer]
Hi,
Sorry for the late reply. You could just list all the maf files and use the function  `marge_mafss()`.

```r
maf_files = list.files(path = "D:\downs\mutation\files", pattern = "*.\\.maf\$", full.names = TRUE)
mymaf = maftools::merge_mafs(mafs = maf_files, verbose = TRUE)
maftools::write.mafSummary(maf = mymaf, basename = "mymaf")
```

Hope this helps.

**Reply by GAUSS66-well** [reply]
wow,it has worked !! thank you so much!

**Reply by amarinderthind** [reply]
This function does not seem to retain sample info but merges all file variants in one sample name after removing duplicate variants.

**Reply by amarinderthind** [reply]
@PoisonAlien Got it supposed to rename, `Source_MAF` to ` Tumor_Sample_Barcode`. But the issue is it removes variants, in the first place, by considering them duplicates (Even though they belong to different samples).
